types/event_bus.go

```diff
@@ -82,7 +82,11 @@ func (b *EventBus) PublishEventVote(vote EventDataVote) error {
 func (b *EventBus) PublishEventTx(tx EventDataTx) error {
 	// no explicit deadline for publishing events
 	ctx := context.Background()
-	b.pubsub.PublishWithTags(ctx, TMEventData{tx}, map[string]interface{}{EventTypeKey: EventTx, TxHashKey: fmt.Sprintf("%X", tx.Tx.Hash())})
+	tags := tx.Tags
+	// add predefined tags (they should overwrite any existing tags)
+	tags[EventTypeKey] = EventTx
```

Comment:
should we check these here or in the execution and log something if the app returned them so it's notified they were overwritten? At least we need to document this well

Reply:
good idea